### PR TITLE
[electron-packager] Add all electron-osx-sign options

### PR DESCRIPTION
**Pull request by ffflorian**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://github.com/electron-userland/electron-osx-sign/blob/master/index.d.ts#L26
  - https://github.com/electron-userland/electron-packager/blob/master/src/mac.js#L10
  - https://github.com/electron-userland/electron-packager/blob/master/src/mac.js#L293
  - https://github.com/electron-userland/electron-packager/blob/master/src/mac.js#L346
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/electron-packager/index.d.ts [hunk 1]
```diff
@@ -57,9 +57,18 @@ declare namespace electronPackager {
     }
 
     interface ElectronOsXSignOptions {
-        identity?: string;
         entitlements?: string;
         "entitlements-inherit"?: string;
+        identity?: string | boolean;
```
Comment (cortopy) [inline types/electron-packager/index.d.ts]: [electron-osx-sign](https://www.npmjs.com/package/electron-osx-sign#opts) docs say that identity is till a string. Could you clarify?

**Review by cortopy — CHANGES_REQUESTED** [review]

File: types/electron-packager/index.d.ts (hunk 1 above)
Comment (ffflorian) [reply types/electron-packager/index.d.ts]: Please see `mac.js` lines 356-361:

https://github.com/electron-userland/electron-packager/blob/92d09bba34599283a794fd6f24b88470f0cb1074/src/mac.js#L356-L361

Comment (cortopy) [reply types/electron-packager/index.d.ts]: awesome, thanks!

**Review by cortopy — APPROVED** [review]

File: types/electron-packager/index.d.ts (hunk 1 above)
Comment (ffflorian) [reply types/electron-packager/index.d.ts]: The more I think about it, it should actually be `string | true`, what do you think?

Comment (cortopy) [reply types/electron-packager/index.d.ts]: Definitely, well spotted. There doesn't seem to be logic for handling `false`. Could you also add an inline comment with the link you shared here for future reference?

Comment (ffflorian) [reply types/electron-packager/index.d.ts]: Done :slightly_smiling_face:

Comment (cortopy) [reply types/electron-packager/index.d.ts]: :+1:

File: types/electron-packager/index.d.ts [hunk 2]
```diff
@@ -57,9 +58,19 @@ declare namespace electronPackager {
...
+        // see https://github.com/electron-userland/electron-packager/blob/92d09bba34599283a794fd6f24b88470f0cb1074/src/mac.js#L359
+        identity?: string | true;
+        "identity-validation"?: boolean;
+        ignore?: string;
+        keychain?: string;
+        "pre-auto-entitlements"?: boolean;
+        "pre-embed-provisioning-profile"?: boolean;
+        "provisioning-profile"?: string;
+        requirements?: string;
+        type?: string;
```
Comment (malept) [inline types/electron-packager/index.d.ts]: Is there a way to say something like

```typescript
import { SignOptions } from 'electron-osx-sign';

// ...
interface ElectronOsxSignOptions extends SignOptions {
  identity?: string | true;
}
```

and have `identity` get overridden properly by TypeScript? Personally I'd rather have as little copypasta as possible.

**Review by malept — CHANGES_REQUESTED** [review]

File: types/electron-packager/index.d.ts (hunk 2 above)
Comment (malept) [reply types/electron-packager/index.d.ts]: Oh, the change I described is in #36441? I'm a little confused.

Comment (ffflorian) [reply types/electron-packager/index.d.ts]: @malept #36441 will be merged after this PR.

I'm also against copypasta, but to import `electron-osx-sign`, https://github.com/microsoft/types-publisher/pull/627 needs to be merged. And this sometimes takes a few days... :slightly_smiling_face:

Comment (cortopy) [reply types/electron-packager/index.d.ts]: @malept is that valid typescript? See this question in [stackoverflow](https://stackoverflow.com/questions/41285211/overriding-interface-property-type-defined-in-typescript-d-ts-file)

What about once the types get published:

```typescript
import { SignOptions } from 'electron-osx-sign';

interface ElectronOsxSignOptions extends Omit<SignOptions, 'identity'>  {
  identity?: string | true;
}
```

Comment (malept) [reply types/electron-packager/index.d.ts]: > is that valid typescript?

I did say "something like" as my TypeScript skills are not very good. @ffflorian answered my question.

**Review by malept — APPROVED** [review]
This is fine given the existence of #36441.

**Pull request merged**